src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -92,15 +92,12 @@ def _replace_cloned_args(clone_args, user_args):
             clone_args[prop] = user_args[prop]
     return
     
-def initialize_clone(request_args, reqmgr_db_service):
+def initialize_clone(request_args, originalArgs, argsDefinition):
     """
     Initialize a Clone arguments by inheriting and overwriting argument from OriginalRequest
     """
-    requests = reqmgr_db_service.getRequestByNames(request_args["OriginalRequestName"])
-    clone_args = requests.values()[0]
-    # TODO: need to validate new overwrite request_args here since it will skip the clone_args validtaiton
-    _replace_cloned_args(clone_args, request_args)
-    clone_args = {k: v for k, v in clone_args.iteritems() if k not in ARGS_TO_REMOVE_FROM_ORIGINAL_REQUEST}
+    _replace_cloned_args(originalArgs, request_args)
+    clone_args = {k: v for k, v in originalArgs.iteritems() if k in argsDefinition}
```

Comment:
Actually, we need the same mechanism as in:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/WMWorkloadTools.py#L292

there must have a smarter way to handle it...